_posts/2021-11-24-custom-properties.md

```diff
@@ -0,0 +1,53 @@
+---
+layout: post
+title: "CSS Custom Properties"
+date: 2021-11-24 12:33:53
+tags:
+- css
+- Sass
+categories:
+- css
+twitter_text: Custom Properties allow you to change the value of a variable in your css or Sass. Find out how.
+authors: "Ted Krueger"
+image:
+image_url:
+image_credit:
+---
+Have you ever wanted to change the value of a Sass variable based on screen size, a state, or for any other reason that might require changing a Sass variable? Well you can’t. Not with a Sass variable anyway, but you can by using a css custom property. These are sometimes called css variables.
+
+There have been times where I have defined a variable in Sass and then wanted to change it when that element is `.active` for example. Something like:
+
+```
+.element {
+  $background: red;
+  background-color: $background;
+
+  &.active {
+    $background: blue;
+  }
+}
+```
+
+ It was always frustrating that I couldn’t do this. Then I found out about custom properties. They work in standard css as well as Sass (scss files). It’s pretty awesome because now you can totally change the value of a variable. By using custom properties, the above code becomes:
+
+```
+.element {
+  --background: red;
+  background-color: var(--background);
+
+  &.active {
+    --background: blue;
+  }
+}
+```
+
+Notice the difference between how the two variables are used. When using a Sass variable, you define it with a `$` whereas with custom properties you need to start them with a double dash (`--`).
+
+Sass variable: `$background: red`
+CSS Custom Property: `--background: red`
+
+You use them in your styles differently as well. Sass variables are more straight forward, you simply add the same variable into the style value but with a custom property you need to wrap it in a `var()`. 
```

Comment:
- Would it be good to explain `var()` a bit? something along the lines of: 
"... to set the value of a CSS property to a CSS custom property, you must return the custom property's value using the CSS `var()` function."

Reply:
Good catch, Ryan. Yeah, I think that would be a good change.

Reply:
I changed your suggestion a bit, @mrtrimble. I don't know if the custom property's value is what we would be adding the the `var` function but rather the variable name.

Reply:
Gotcha, yeah I think you made it more succinct!